### ISSUE TITLE
Checked C compilers causes -Wnull-pointer-arithmetic warning 

### ISSUE THREAD

**Issue by nmeum** [Microsoft/checkedc-clang#457]
I believe I found a bug in the Checked C compiler which can be reproduced with the following minimal example code:

```C
#include <stdlib.h>
#include <sys/types.h>

#pragma BOUNDS_CHECKED ON

static void myfunc(_Array_ptr<void> data : byte_count(size),
                                        size_t size)
{
	(void)data;
}

int main(void)
{
	myfunc(NULL, 5);
	return 0;
}
```

When compiling this code snipped with `-Wextra` clang yields the following warning:

```
$ clang -Wextra -o test test.c
warning: arithmetic on a null pointer treated as a cast from integer to pointer is a GNU extension
      [-Wnull-pointer-arithmetic]
1 warning generated.
$ clang --version
clang version 6.0.0 (https://github.com/Microsoft/checkedc-clang 9533284945156bb6ff2c2f64bfdaaaa84b361757) (https://github.com/Microsoft/checkedc-llvm d8d78c7a13472b344e429d1ae320bc4f01ecb6db)                                              
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/checkedc-llvm/bin
```

Since the warning doesn't include any line information I assume that this warning is caused by runtime code generate for the bounds check. If my assumptions are correct I would consider this a bug since it should imho be possible to compile Checked C code with `-Wextra`.

**BTW:** I couldn't reproduce this error with the Checked C compiler version I used previously, the `--version` output of that previous compiler version I used looks as follows:

```
clang version 6.0.0 (https://github.com/Microsoft/checkedc-clang ff5fa0c2c71806b2aee7825d986aa3abe4d027f1) (https://github.com/Microsoft/checkedc-llvm 2135ac802652eece135d175e5d23ac0145de9acd)                                              
Target: x86_64-unknown-linux-gnu                                                                                                                                                                                                              
Thread model: posix
InstalledDir: /opt/checkedc-llvm-bkp/bin
```

**Comment by dtarditi** [comment]
Yes, this is a bug.  

I think I know what's causing this.   To check bounds declarations for calls, at each parameter with bounds, we substitute the actual arguments for the parameters that occur in the bounds. When the actual argument is null, we create an expression that satisfies the criteria for the warning.   I'll have to figure out why and when this expression is being checked.   This is a compiler-generated expression, so we really don't want warnings about it.

**Comment by dtarditi** [comment]
This is fixed by PR #486.
